reader.go

```diff
@@ -783,6 +787,55 @@ func (r *Reader) commitLoop(conn *Conn) func(stop <-chan struct{}) {
 	}
 }
 
+// partitionWatcher queries kafka and watches for partition changes, triggering a rebalance if changes are found.
```

Comment:
If you can leave a little bit of clarifying documentation as to why it's okay to return from errors here, that would be great.  On an initial reading, I had thought that returning in the face of an error might not be appropriate because it could introduce consumer group rebalance churn due to tangential errors (e.g. failure to list partitions isn't necessarily a good reason to rebalance the group...it would be a more cautious approach to let the group continue and retry in the background).  However, since we're share the `Conn` with the other CG go routines, it means that we're interacting the CG coordinator...and if we lose that connection, then the other go routines in the `runGroup` will also be exiting and we'll have a rebalance on our hands.  Do you see where I'm going with that?  If you can find a way to say it more concisely, that'd be great. 😄